### PR TITLE
Implement interface to print runtime information

### PR DESCRIPTION
**Pull request by kmaehashi**
Implements interface to print runtime information. Took over #4298.
Requires https://github.com/cupy/cupy/pull/1067.

~~TODO: add test.~~ (done)

### CODE REVIEW

**Comment by kmaehashi** [comment]
Added tests.

### Output examples

When CuPy is unavailable:

```
>>> import chainer
>>> chainer.print_runtime_info()
Chainer: 4.0.0rc1
NumPy: 1.14.2
CuPy: Not Available
```

When CuPy is available:

```
>>> import chainer
>>> chainer.print_runtime_info()
Chainer: 4.0.0rc1
NumPy: 1.14.2
CuPy:
  CuPy Version          : 4.0.0rc1
  CUDA Root             : /usr/local/cuda
  CUDA Build Version    : 9000
  CUDA Driver Version   : 9000
  CUDA Runtime Version  : 9000
  cuDNN Build Version   : 7100
  cuDNN Version         : 7100
  NCCL Build Version    : 2102
```

File: chainer/_runtime_info.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+def get_runtime_info():
+    return _RuntimeInfo()
+
+
+def print_runtime_info(out=None):
+    if out is None:
+        out = sys.stdout
+    out.write(str(get_runtime_info()))
+    out.flush()
```
Comment (niboshi) [inline chainer/_runtime_info.py]: `flush` method may not exist.
It's better to either avoid it or test its existence first.

Comment (niboshi) [reply chainer/_runtime_info.py]: There's no official specification of file-like objects.
You can define an object with only `write` method and call it a file-like.

Comment (kmaehashi) [reply chainer/_runtime_info.py]: OK, I fixed to first check existence of the flush method.
(It seems other features (`CupyMemoryProfileHook`, `ProgressBar`, etc...) also assumes `flush`.)

Comment (niboshi) [reply chainer/_runtime_info.py]: I see. It's better to fix them as well.

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**